File: MPAndroidChartJava/app/src/main/java/com/example/mpandroidchartjava/BarChartActivity.java
package com.example.mpandroidchartjava;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.github.mikephil.charting.charts.BarChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.components.YAxis;
import com.github.mikephil.charting.data.BarData;
import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.formatter.ValueFormatter;
import com.github.mikephil.charting.utils.ColorTemplate;

import java.util.ArrayList;

public class BarChartActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bar_chart);

        BarChart chart = findViewById(R.id.barchart);

        ArrayList<BarEntry> NoOfEmp = new ArrayList<BarEntry>();

        NoOfEmp.add(new BarEntry(945f, 0));
        NoOfEmp.add(new BarEntry(1040f, 1));
        NoOfEmp.add(new BarEntry(1133f, 2));
        NoOfEmp.add(new BarEntry(1240f, 3));
        NoOfEmp.add(new BarEntry(1369f, 4));
//        NoOfEmp.add(new BarEntry(1487f, 5));
//        NoOfEmp.add(new BarEntry(1501f, 6));
//        NoOfEmp.add(new BarEntry(1645f, 7));
//        NoOfEmp.add(new BarEntry(1578f, 8));
//        NoOfEmp.add(new BarEntry(1695f, 9));

        XAxis xAxis = chart.getXAxis();
        xAxis.enableGridDashedLine(10f, 10f, 2f);
        xAxis.setDrawGridLines(true);
        xAxis.setDrawLabels(true);
        xAxis.setDrawAxisLine(true);

        xAxis.setPosition(XAxis.XAxisPosition.BOTTOM);
        xAxis.setSpaceMax(0.5f);
        xAxis.setSpaceMin(0.5f);
        xAxis.setDrawGridLines(true);
        xAxis.setDrawAxisLine(true);
        xAxis.setAxisLineColor(getResources().getColor(R.color.teal_200));

//        YAxis yAxis = chart.getAxisLeft();
//        yAxis.setEnabled(true);
//        yAxis.setAxisMinimum(1f);
//        yAxis.setAxisMaximum(20f);
//
//        yAxis.enableAxisLineDashedLine(10f, 10f, 10f);
//        yAxis.setDrawGridLines(true);
//        yAxis.setGranularityEnabled(true);
//        yAxis.enableGridDashedLine(10f, 10f, 0f);
//        yAxis.setDrawAxisLine(true);
//
//        YAxis yAxis1 = chart.getAxisLeft();
//        yAxis1.enableAxisLineDashedLine(10f, 10f, 10f);
//        yAxis1.setDrawGridLines(true);
//        yAxis1.setDrawAxisLine(true);


        BarDataSet bardataset = new BarDataSet(NoOfEmp, "No Of Employee");
        chart.animateY(5000);
        BarData data = new BarData(bardataset);
        bardataset.setColors(ColorTemplate.COLORFUL_COLORS);
        chart.setData(data);

    }
}

File: MPAndroidChart/app/src/main/java/com/example/mpandroidchart/BarChartActivity.kt
package com.example.mpandroidchart

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.github.mikephil.charting.charts.BarChart
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.components.YAxis
import com.github.mikephil.charting.data.BarData
import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry


class BarChartActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bar_chart)

        val chart: BarChart = findViewById(R.id.barchart)

        val NoOfEmp: MutableList<BarEntry> = mutableListOf()

        NoOfEmp.add(BarEntry(9f, 0f))
        NoOfEmp.add(BarEntry(10f, 1f))
        NoOfEmp.add(BarEntry(11f, 2f))
        NoOfEmp.add(BarEntry(12f, 3f))
        NoOfEmp.add(BarEntry(13f, 4f))

        val NoOfEmp2: MutableList<BarEntry> = mutableListOf()

        NoOfEmp2.add(BarEntry(9f, 2f))
        NoOfEmp2.add(BarEntry(10f, 4f))
        NoOfEmp2.add(BarEntry(11f, 1f))
        NoOfEmp2.add(BarEntry(12f, 3f))
        NoOfEmp2.add(BarEntry(13f, 0f))

        val year: MutableList<String> = mutableListOf()

        year.add("2008")
        year.add("2009")
        year.add("2010")
        year.add("2011")
        year.add("2012")
        year.add("2013")
        year.add("2014")
        year.add("2015")
        year.add("2016")
        year.add("2017")

        val bardataset = BarDataSet(NoOfEmp, "No Of Employee") // object data
        val bardataset2 = BarDataSet(NoOfEmp2, "No") // object data
        bardataset2.setColors(Color.RED)
        chart.animateY(1000) // hieu ung khi chay du lieu

        /* config bar chart */
        chart.description.isEnabled = false
        chart.setDrawBarShadow(false)
        chart.legend.isEnabled = false
        chart.setScaleEnabled(true)
        chart.setPinchZoom(false)
        chart.setFitBars(false)

        chart.axisLeft.minWidth = 3f
        chart.setVisibleXRange(1f, 10f)
        chart.axisLeft.setPosition(YAxis.YAxisLabelPosition.OUTSIDE_CHART)
        chart.axisLeft.setDrawGridLines(true)
        chart.axisLeft.enableGridDashedLine(10f, 10f, 0f)
        chart.axisLeft.axisMinimum = 0f // set data start from bottom  = 0f
//        chart.axisLeft.axisMaximum = 0f
//        chart.axisRight.axisMaximum = 0f
        chart.axisRight.isEnabled = false
        chart.xAxis.setDrawGridLines(true)
        chart.xAxis.enableGridDashedLine(10f, 10f, 0f)
        chart.xAxis.position = XAxis.XAxisPosition.BOTTOM
        chart.xAxis.axisLineColor = Color.RED
        /* **************** */

        val data = BarData(bardataset, bardataset2) // put into bar data
        chart.data = data
    }
}